File: app/src/main/java/com/example/travellist/TravelInfo.java
package com.example.travellist;

import android.os.Bundle;

/**
 * Esta clase contiene la informacion de un viaje. Los datos que almacena son la ciudad, el pais,
 * el anyo del viaje y una nota opcional.
 *
 *
 */
public class TravelInfo {

	private String city;
	private String country;
	private int year;
	private String note;

	public static TravelInfo fromBundle(Bundle bundle) {
		return new TravelInfo(bundle);
	}
	
	public TravelInfo(String city, String country, int year, String note){
		this.city = city;
		this.country = country;
		this.year = year;
		this.note = note;
	}
	
	public TravelInfo(String city, String country, int year){
		this(city, country, year, null);
	}

	public TravelInfo(Bundle info) {
		this(
				info.getString("City"),
				info.getString("Country"),
				info.getInt("Year"),
				info.getString("Note"));
	}

	public Bundle toBundle() {
		Bundle info = new Bundle();
		info.putString("City" , this.city);
		info.putString("Country" , this.country);
		info.putInt("Year" , this.year);
		info.putString("Note" , this.note);
		return info;
	}

	public String getCity() {
		return city;
	}
	
	public void setCity(String city) {
		this.city = city;
	}
	
	public String getCountry() {
		return country;
	}
	
	public void setCountry(String country) {
		this.country = country;
	}
	
	public int getYear() {
		return year;
	}
	
	public void setYear(int year) {
		this.year = year;
	}
	
	public void setNote(String note) {
		this.city = note;
	}
	
	public String getNote() {
		return note;
	}
	
	
	
}

File: app/src/main/java/com/example/travellist/TravelInfo.kt
package com.example.travellist

import android.os.Bundle

/**
 * Esta clase contiene la informacion de un viaje. Los datos que almacena son la ciudad, el pais,
 * el anyo del viaje y una nota opcional.
 *
 *
 */
class TravelInfo @JvmOverloads constructor(var city: String?, var country: String?, var year: Int, var note: String? = null) {

    constructor(info: Bundle) : this(
            info.getString(KEY_CITY),
            info.getString(KEY_COUNTRY),
            info.getInt(KEY_YEAR),
            info.getString(KEY_NOTE))

    fun toBundle(): Bundle {
        val info = Bundle()
        info.putString(KEY_CITY, this.city)
        info.putString(KEY_COUNTRY, this.country)
        info.putInt(KEY_YEAR, this.year)
        info.putString(KEY_NOTE, this.note)
        return info
    }

    companion object {

        const val KEY_INFO = "Info"
        const val KEY_POSITION = "Position"
        const val KEY_CITY = "City"
        const val KEY_COUNTRY = "Country"
        const val KEY_YEAR = "Year"
        const val KEY_NOTE = "Note"

        fun fromBundle(bundle: Bundle): TravelInfo {
            return TravelInfo(bundle)
        }
    }


}
